From 3f45fb0ed79baf70c4afa90c45b8b9773b39b77e Mon Sep 17 00:00:00 2001
From: Zhi Wang <zhi.a.wang@intel.com>
Date: Thu, 26 Mar 2015 21:24:05 +0800
Subject: [PATCH 028/403] vgt: try to dump ring buffer from apeature if
 vgt_gma_to_va() fail.

This patch can fix BUG 736#.

Signed-off-by: Zhi Wang <zhi.a.wang@intel.com>
---
 drivers/gpu/drm/i915/vgt/utility.c |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/gpu/drm/i915/vgt/utility.c b/drivers/gpu/drm/i915/vgt/utility.c
index 7945f87..87074cb 100644
--- a/drivers/gpu/drm/i915/vgt/utility.c
+++ b/drivers/gpu/drm/i915/vgt/utility.c
@@ -278,7 +278,14 @@ void common_show_ring_buffer(struct pgt_device *pdev, int ring_id, int bytes,
 	p_head &= RB_HEAD_OFF_MASK;
 	ring_len = _RING_CTL_BUF_SIZE(p_ctl);
 	p_contents = vgt_gma_to_va(vgt->gtt.ggtt_mm, p_start);
+	if (!p_contents) {
+		if (pdev->enable_execlist)
+			return;
 
+		printk("Looks this ring buffer doesn't belong to current render owner.\n");
+		printk("Try to dump it from aperture.\n");
+		p_contents = phys_aperture_vbase(pdev) + p_start;
+	}
 #define WRAP_OFF(off, size)			\
 	({					\
 		u64 val = off;			\
-- 
1.7.10.4

